### PR TITLE
Show a prompt to reopen the workspace if devices aren't found

### PR DESCRIPTION
**Pull request by helin24**
This is specifically for an internal project that requires users to start/restart a process on their local machine to expose devices.

### CODE REVIEW

File: src/extension/extension.ts [hunk 1]
```diff
@@ -259,7 +260,15 @@ export async function activate(context: vs.ExtensionContext, isRestart: boolean
 	// Fire up Flutter daemon if required.
 	if (workspaceContext.hasAnyFlutterProjects && sdks.flutter) {
 		flutterDaemon = new FlutterDaemon(logger, workspaceContext as FlutterWorkspaceContext, flutterCapabilities);
-		deviceManager = new FlutterDeviceManager(logger, flutterDaemon, config, workspaceContext);
+
+		let runIfNoDevices;
+		if (workspaceContext.config.forceFlutterWorkspace && workspaceContext.config.restartMacDaemonMessage) {
+			runIfNoDevices = () => {
+				const instruction = workspaceContext.config.restartMacDaemonMessage;
+				promptToReloadExtension(`${instruction} (VSCode settings currently expects port: ${config.daemonPort}.)`, `Reopen this workspace`);
```
Comment (DanTup) [inline src/extension/extension.ts:268]: nit: I think this reads better as "VSCode settings currently expect port" (no "s" on "expect"), or "VSCode currently expects port" (no "settings"). VS Code is also usually written with a space or just as "Code". Perhaps just "(Settings current expect port x)" also works and is a bit less verbose.

File: src/shared/vscode/device_manager.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export class FlutterDeviceManager implements vs.Disposable {
 	private emulators: Emulator[] = [];
 	private readonly knownEmulatorNames: { [key: string]: string } = {};
 
-	constructor(private readonly logger: Logger, private daemon: IFlutterDaemon, private readonly config: { flutterCustomEmulators: CustomEmulatorDefinition[], flutterSelectDeviceWhenConnected: boolean, flutterShowEmulators: "local" | "always" | "never", projectSearchDepth: number }, private readonly workspaceContext: WorkspaceContext) {
+	constructor(private readonly logger: Logger, private daemon: IFlutterDaemon, private readonly config: { flutterCustomEmulators: CustomEmulatorDefinition[], flutterSelectDeviceWhenConnected: boolean, flutterShowEmulators: "local" | "always" | "never", projectSearchDepth: number, daemonPort: number | undefined }, private readonly workspaceContext: WorkspaceContext, runIfNoDevices?: () => void) {
```
Comment (DanTup) [inline src/shared/vscode/device_manager.ts]: Is the `, daemonPort: number | undefined` added here needed?

**Review by DanTup — APPROVED** [review]

File: src/shared/vscode/device_manager.ts (hunk 2 above)
Comment (DanTup) [reply src/shared/vscode/device_manager.ts]: (this is already fixed)

File: src/extension/extension.ts (hunk 1 above)
Comment (helin24) [reply src/extension/extension.ts:268]: argh i've been writing "VSCode" everywhere haha 😢

File: src/shared/vscode/device_manager.ts (hunk 2 above)
Comment (helin24) [reply src/shared/vscode/device_manager.ts]: oops meant to delete that

**Pull request merged**